internal/api/assignments.go

```diff
@@ -110,21 +148,40 @@ func (r *Router) assignmentDelete(c echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, "error parsing request body").SetInternal(err)
 	}
 
-	subjID, err := gidx.Parse(reqBody.SubjectID)
+	assigneeID, err := gidx.Parse(reqBody.SubjectID)
 	if err != nil {
 		return echo.NewHTTPError(http.StatusBadRequest, "error parsing subject ID").SetInternal(err)
 	}
 
-	subjResource, err := r.engine.NewResourceFromID(subjID)
+	assigneeResource, err := r.engine.NewResourceFromID(assigneeID)
 	if err != nil {
 		return echo.NewHTTPError(http.StatusBadRequest, "error parsing resource type from subject").SetInternal(err)
 	}
 
+	subjectResource, err := r.currentSubject(c)
+	if err != nil {
+		return err
+	}
+
+	roleResource, err := r.engine.NewResourceFromID(roleID)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "error getting resource").SetInternal(err)
+	}
+
+	resource, err := r.engine.GetRoleResource(ctx, roleResource, "")
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "error getting resource").SetInternal(err)
+	}
+
+	if err := r.checkActionWithResponse(ctx, subjectResource, actionRoleUpdate, resource); err != nil {
```

Comment:
```suggestion
	if err := r.checkActionWithResponse(ctx, subjectResource, actionRoleDelete, resource); err != nil {
```

Reply:
I went back and forth on this one. In this case, we're not deleting the role itself - rather, we're removing an assignment for the role. That's why I kept it as `actionRoleUpdate`.